### PR TITLE
feat: add help section helper

### PR DESCRIPTION
**Pull request by peternhale**
@W-9791511@

### CODE REVIEW

File: src/util/helpSectionUtils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { HelpSection } from '@oclif/core';
+import { ORG_CONFIG_ALLOWED_PROPERTIES, OrgConfigProperties } from '../org/orgConfigProperties';
+import { SFDX_ALLOWED_PROPERTIES, SfdxPropertyKeys } from '../config/config';
+import { EnvironmentVariable, SUPPORTED_ENV_VARS } from '../config/envVars';
+
+export const toHelpSection = (
```
Comment (peternhale) [inline src/util/helpSectionUtils.ts:12]: produces a help section from the provided array of org/sfdx/env var enums

**Review by mdonnalley — CHANGES_REQUESTED** [review]
I think this is too specific to `sf` to belong in sfdx-core - is there a reason we wouldn't just put this in the custom help class?

**Comment by peternhale** [comment]
> I think this is too specific to `sf` to belong in sfdx-core - is there a reason we wouldn't just put this in the custom help class?

@mdonnalley placing this in help class would require that each command reference that class. I could make this available in @salesforce/command.

**Comment by mdonnalley** [comment]
> > I think this is too specific to `sf` to belong in sfdx-core - is there a reason we wouldn't just put this in the custom help class?
> 
> @mdonnalley placing this in help class would require that each command reference that class. I could make this available in @salesforce/command.

@peternhale yeah, I think that would make more sense

**Comment by peternhale** [comment]
> > > I think this is too specific to `sf` to belong in sfdx-core - is there a reason we wouldn't just put this in the custom help class?
> > 
> > 
> > @mdonnalley placing this in help class would require that each command reference that class. I could make this available in @salesforce/command.
> 
> @peternhale yeah, I think that would make more sense

@mdonnalley moving this to @salesforce/command is complicated, since this code needs access to core v3 and command is dependent on v2. I can alias in v3, but seem pretty heavy at the moment.

File: package.json [hunk 2]
```diff
@@ -39,6 +39,7 @@
     }
   },
   "dependencies": {
+    "@oclif/core": "^0.5.35",
```
Comment (mdonnalley) [inline package.json:42]: this should be removed

**Review by mdonnalley — CHANGES_REQUESTED** [review]

**Review by mdonnalley — APPROVED** [review]

**Pull request merged**